sway/commands/fullscreen.c

```diff
@@ -18,16 +18,13 @@ struct cmd_results *cmd_fullscreen(int argc, char **argv) {
 		return cmd_results_new(CMD_FAILURE,
 				"Can't run this command while there's no outputs connected.");
 	}
-	struct sway_node *node = config->handler_context.node;
 	struct sway_container *container = config->handler_context.container;
-	struct sway_workspace *workspace = config->handler_context.workspace;
-	if (node->type == N_WORKSPACE && workspace->tiling->length == 0) {
-		return cmd_results_new(CMD_FAILURE,
-				"Can't fullscreen an empty workspace");
-	}
 
-	// If in the scratchpad, operate on the highest container
-	if (container && !container->pending.workspace) {
+	if (!container) {
+		// If the focus is not a container, do nothing successfully
+		return cmd_results_new(CMD_SUCCESS, NULL);
```

Comment:
Yes it is, based on a quick experiment in i3; in a workspace with a lone terminal I tried:

```
$ i3-msg focus parent, fullscreen enable
[{"success":true},{"success":true}]
```

I looked into it again and i3 seems to consider all active workspaces as fullscreened, and inactive workspaces as not fullscreened. So when ws1 is focused,

```
$ i3-msg -t get_tree | jq -rc 'recurse(.nodes[]) | select(.type == "workspace") | {name: .name, fullscreen: .fullscreen_mode}'
{"name":"__i3_scratch","fullscreen":1}
{"name":"1","fullscreen":1}
{"name":"2","fullscreen":0}
```

which sort of explains why this is a success: it's redundant. So it's in-line with the behavior:

```
$ swaymsg -t fullscreen enable | jq -c
[{"success":true}]
$ swaymsg -t fullscreen enable
[{"success":true}]
```

Atm sway always reports workspaces as fullscreened, so I guess that should probably be changed as well: https://github.com/swaywm/sway/blob/a6dc829ed00585755adeb25a18373b163f9a94c9/sway/ipc-json.c#L429

Reply:
Does `!container` always mean that we're dealing with a workspace? Maybe something like `config->handler_context.node->type == N_WORKSPACE` would be safer? (And return an error otherwise)

Reply:
Criteria commands always match a container (if they execute), so `!container` means the seat focus is not a container, so either a workspace or NULL. Returning an error in that case would make sense, but notice that there doesn't appear to be any case where i3 returns an error for the fullscreen command:

https://github.com/i3/i3/blob/fcae64f7fd97c02f36200a4c60ab7ec2fd4410df/src/commands.c#L1510

I'm not sure if that should be considered for compatibility.